checkers_bot_tournament/game.py

```diff
@@ -105,20 +113,22 @@ def import_pdn(self, filename: str) -> None:
                         f"Invalid move in import_pdn for colour: {str(self.current_turn)}, turn: {self.move_number + 1}, move: {move}"
                     )
 
-            self.move_piece(move_obj, True)
+            result, _ = self.execute_move(move_obj, True)
+            # Don't accept pdns where the game has already finished
+            if result:
+                raise RuntimeError(
+                    f"PDN game: {self.pdn} is already complete on move {self.move_number} with result: {result.name}. There's nothing for bots to do!"
+                )
             self.swap_turn()
 
-        # Check that the game has NOT ended at this point
-        if not self.board.get_move_list(self.current_turn):
-            raise RuntimeError(f"PDN game: {self.pdn} is already complete! Nothing for bots to do!")
-
-    def move_piece(self, move: Move, from_import: bool = False) -> None:
-        """
-        Only used by import_pdn and for testing purposes.
-        """
+    def execute_move(
+        self, move: Move, from_import: bool = False
+    ) -> tuple[Optional[Result], list[Move]]:
```

Comment:
Hmm it seems like you only either return a result or list of moves, not both, so you could use a `Union[Result, list[Move]]` and just return one or the other. 

Then use isinstance to get which one was returned. This enforces the fact that only one of these will be returned.